x-pack/test/functional/page_objects/graph_page.ts

```diff
@@ -0,0 +1,275 @@
+      await find.clickByCssSelector('.gphIndexSelect > option[label="' + pattern + '"]');
+    }
+
+    async clickAddField() {
+      await retry.try(async () => {
+        await find.clickByCssSelector('#addVertexFieldButton');
+        // make sure the fieldSelectionList is not hidden
+        await testSubjects.exists('fieldSelectionList');
+      });
+    }
+
+    async selectField(field: string) {
+      await find.clickDisplayedByCssSelector(
+        'select[id="fieldList"] > option[label="' + field + '"]'
+      );
+      await find.clickDisplayedByCssSelector('button[ng-click="addFieldToSelection()"]');
+    }
+
+    async addField(field: string) {
+      log.debug('click Add Field icon');
+      await this.clickAddField();
+      log.debug('select field ' + field);
+      await this.selectField(field);
+    }
+
+    async query(str: string) {
+      await find.setValue('input.kuiLocalSearchInput', str);
+      await find.clickDisplayedByCssSelector('button.kuiLocalSearchButton');
+    }
+
+    private getPositionAsString(x: string, y: string) {
+      return `${x}-${y}`;
+    }
+
+    private async getCirclePosition(element: WebElementWrapper) {
+      const x = await element.getAttribute('cx');
+      const y = await element.getAttribute('cy');
+      return this.getPositionAsString(x, y);
+    }
+
+    private async getLinePositions(element: WebElementWrapper) {
+      const x1 = await element.getAttribute('x1');
+      const y1 = await element.getAttribute('y1');
+      const x2 = await element.getAttribute('x2');
+      const y2 = await element.getAttribute('y2');
+      return [this.getPositionAsString(x1, y1), this.getPositionAsString(x2, y2)];
+    }
+
+    async isolateEdge(edge: Edge) {
+      const from = edge.sourceNode.label;
+      const to = edge.targetNode.label;
+
+      // select all nodes
+      await testSubjects.click('graphSelectAll');
+
+      // go through all nodes and remove every node not source or target
+      const selections = await find.allByCssSelector('.gphSelectionList__field');
+      for (const selection of selections) {
+        const labelElement = await selection.findByTagName('span');
+        const selectionLabel = await labelElement.getVisibleText();
+        log.debug('Looking at selection ' + selectionLabel);
+        if (selectionLabel !== from && selectionLabel !== to) {
+          (await selection.findByClassName('gphNode__text')).click();
+          await PageObjects.common.sleep(200);
+        }
+      }
+
+      // invert selection to select all nodes not source or target
+      await testSubjects.click('graphInvertSelection');
+
+      // remove all other nodes
+      await testSubjects.click('graphRemoveSelection');
+    }
+
+    async clickEdge(edge: Edge) {
+      await this.stopLayout();
+      await PageObjects.common.sleep(1000);
+      await edge.element.click();
+      await this.startLayout();
+    }
+
+    async selectLinkedNodes() {
```

Comment:
I think `selectLinkedNodes` is currently unused and can be removed

Reply:
Good catch, cleaned it up